### PR TITLE
Preorder Traversal in C

### PR DESCRIPTION
**Pull request by rohangawhade**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes
### Description

Added preorder traversal using recursion and iteration in C
### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#415

### CODE REVIEW

**Comment by siddharth25pandey** [comment]
@rohangawhade Code is Incomplete, Send a new **PR**

File: C/ds/Preordertraversal.c [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+ struct BST *left;
+ struct BST *right;
+};
+
+struct BST* create();
+void insert(struct BST*, struct BST*);
+void preorderRecursive(struct BST*);
+void preorderIterative(struct BST*);
+
+int main(){
```
Comment (siddharth25pandey) [inline C/ds/Preordertraversal.c:16]: Display message before & after of  inserting element in the node

File: C/ds/Preordertraversal.c [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+		        printf("%d ", root->data);
+				node->right=root;
+				root=root->left;
+			}
+        }
+        else{
+		    printf("%d ", root->data);
+			root=root->right;
+        }
+    }
```
Comment (siddharth25pandey) [inline C/ds/Preordertraversal.c:102]: Add Sample Input & Output

File: C/ds/Preordertraversal.c (hunk 1 above)
Comment (rohangawhade) [reply C/ds/Preordertraversal.c:16]: Added display message.

File: C/ds/Preordertraversal.c (hunk 2 above)
Comment (rohangawhade) [reply C/ds/Preordertraversal.c:102]: Added sample input and output.

**Review by siddharth25pandey — APPROVED** [review]

**Review by vatsalkesarwani12 — APPROVED** [review]

**Review by atarax665 — APPROVED** [review]

**Comment by atarax665** [comment]
@all-contributors please add @rohangawhade  for code.

**Pull request merged**